### ISSUE TITLE
Can't save settings in new urvanov build

### ISSUE THREAD

**Issue by stigzler** [aramk/crayon-syntax-highlighter#463]
I know that there are a lot of work. But, at least, it will work for me...

_Originally posted by @urvanov-ru in https://github.com/aramk/crayon-syntax-highlighter/issues/452#issuecomment-471763264_

I couldn't find any way to get in contact, urvanov - so hopefully this will ping you. Firstly - thanks for bringing this back to life! The best syntax highlighter for WP. 

In you most recent beta (Version _2.8.5_beta - downloaded 14.7.19) there appear to be some settings options missing (choosing default language etc) and also you cannot save settings (no button).

Weirdly, I have an older beta of yours on another site (appears in plugins as "Urvanov Syntax Highlighter" which works better - all options available + can save. 

Just to let you know.

**Comment by urvanov-ru** [comment]
Do you use gutenberg-support brach from https://github.com/urvanov-ru/crayon-syntax-highlighter/tree/gutenberg-support? It should work. I will try to check it in the evening.

Branch **rebranding-refactoring-best-practices** is not ready now.

**Comment by urvanov-ru** [comment]
> In you most recent beta (Version _2.8.5_beta - downloaded 14.7.19) there appear to be some settings options missing (choosing default language etc) and also you cannot save settings (no button).

@stigzler Yes, you are right. 2.8.5 version is from branch rebranding-refactoring-best-practices. I am in the middle of refactoring and fixing. *aramk* created a good plugin. But now it does not follow current Wordpress Plugin Developer Best Practices. Three months ago I tried to upload my fixed version to wordpress plugin repository. They rejected it and wrote how to fix with [the link](https://developer.wordpress.org/plugins/plugin-basics/best-practices/). I am fixing the plugin now in branch *rebranding-refactoring-best-practices*. I will upload it again after some time.

But you can use *gutenberg-support* branch. It is old branch but there no refactoring. Only fixes for Gutenberg.

**Comment by stigzler** [comment]
Aha! That was it. So there's two new versions:

https://github.com/urvanov-ru/crayon-syntax-highlighter
https://github.com/urvanov-ru/crayon-syntax-highlighter/tree/gutenberg-support

That's what caused my confusion. Have downloaded the Gutenberg build and confirm working. :)

Just to let you know that the vbnet lang files are missing from this build. I just lifted them from the original CSH and dropped them in the folder and working fine again. I know, I know - vb.net, right? But great for a hobby coder like me 👍 

I've attached it in case of any use. 

Thanks for doing this, man. Crayon is by way and far the best syntax highlighter in WP and you're keeping it alive! What's the best way to let you know of any bugs or issues on your new builds?

[vbnet.zip](https://github.com/aramk/crayon-syntax-highlighter/files/3393785/vbnet.zip)
